core/src/main/java/aima/core/search/api/BidirectionalProblem.java

```diff
@@ -0,0 +1,20 @@
+package aima.core.search.api;
+
+/**
+ * @author manthan
+ */
+public interface BidirectionalProblem<A, S> extends
+        // S initialState()
+        InitialStateFunction<S>,
+        // List<A> actions(S s)
+        ActionsFunction<A, S>,
+        // List<S> results(S s, A a)
+        ResultFunction<A, S>,
+        // boolean isGoalState(S s)
+        GoalTestPredicate<S>,
+        // double stepCost(S s, A a, S s')
+        StepCostFunction<A, S> {
+    Problem<A,S> getOriginalProblem();
```

Comment:
I did this because :-
1.newChildNode() function requires a problem, but I have BidirectionalProblem as the argument of apply(). Hence to create a node for such case, I modified it accordingly. Could you suggest a better approach.
1. If BidirectionalProblem isn't implemented as above, we won't have functions like initialState(). Thus to make it just like other types of problems like Problem, NondeterministicProblem etc I did so.